.travis.yml

```diff
@@ -0,0 +1,30 @@
+language: cpp
+compiler: gcc
+sudo: require
+dist: trusty
+
+script:
+  - ./autogen.sh # Undocumented
+  - ./configure --prefix=/usr
+  - make -j$(nproc)
+  - make install DESTDIR=$(readlink -f appdir) ; find appdir/
+  - rm -rf appdir/usr/include/ appdir/usr/lib/pkgconfig/ appdir/usr/lib/*.{a,la} # Don't bundle developer stuff
+  - sudo mv appdir/usr/lib/* /usr/local/lib/ # Workaround for https://github.com/probonopd/linuxdeployqt/issues/160
+  - mkdir -p appdir/usr/share/icons/hicolor/128x128/apps ; cp dist/qpdf.png appdir/usr/share/icons/hicolor/128x128/apps/
+  - mkdir -p appdir/usr/share/applications ; cp dist/qpdf.desktop appdir/usr/share/applications/
+  - wget -c "https://github.com/probonopd/linuxdeployqt/releases/download/continuous/linuxdeployqt-continuous-x86_64.AppImage"
+  - chmod a+x linuxdeployqt*.AppImage
+  - unset QTDIR; unset QT_PLUGIN_PATH ; unset LD_LIBRARY_PATH
+  - export LD_LIBRARY_PATH=/usr/local/lib/ # Workaround for https://github.com/probonopd/linuxdeployqt/issues/160
+  - ./linuxdeployqt*.AppImage ./appdir/usr/share/applications/*.desktop -bundle-non-qt-libs
+  - # In addition to the main executable, we have additional ones to process
+  - # ./linuxdeployqt*.AppImage ./appdir/usr/bin/fix-qdf -bundle-non-qt-libs # Not an ELF?
+  - ./linuxdeployqt*.AppImage ./appdir/usr/bin/zlib-flate -bundle-non-qt-libs
+  - # Generate AppImage
+  - ./linuxdeployqt*.AppImage --appimage-extract
+  - rm ./appdir/AppRun ; cp dist/AppRun appdir/ ; chmod a+x ./appdir/AppRun # Replace symlink with custom script
+  - PATH=./squashfs-root/usr/bin:$PATH ./squashfs-root/usr/bin/appimagetool ./appdir/
```

Comment:
There's an awful lot of boilerplate here. Is this just a reflection of the state of how one builds AppImages? This is not something I'd ever heard of. Over time, I'd like to make sure that this continues to work, possibly by setting up a way of running portions of qpdf's test suite using the AppImage. I see your AppRun script handles symlinks whose names match another binary. That's good.

Reply:
I guess my impulse would be to have a script that builds the AppImage, to invoke the script as a make target, and to have the .travis.yml file invoke `make AppImage`. In my day job, I avoid having lots of steps in any CI job. That would also make it easier for me to build an AppImage and test it locally, build a signed AppImage, create an ad-hoc one for giving to someone to test a fix, etc.

Reply:
Normally this would be done automatically by `linuxdeployqt` in a step further above, but since we are using a custom `AppRun` script to make the symlink trick work, we have to manually run the underlying `appimagetool` (which is normally used by `linuxdeployqt`). 
But actually your comment makes me think that we would use the symlink trick for our own linuxdeployqt.AppImage as well, which would allow us to call `appimagetool` without having to extract the AppImage first. I have opened a [feature request](https://github.com/probonopd/linuxdeployqt/issues/175) on it.

Reply:
> I guess my impulse would be to have a script that builds the AppImage, to invoke the script as a make target, and to have the .travis.yml file invoke make AppImage. 

Converting the appropriate lines from `.travis.yml` is something you would have to do because I am not accustomed to writing Makefiles.